### PR TITLE
[V2] chore: Merge master to main v2

### PR DESCRIPTION
**Pull request by mukhoakash**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: charts/index.yaml [hunk 1]
```diff
@@ -3,34 +3,61 @@ entries:
...
     appVersion: latest-v2
-    created: "2022-02-28T11:03:23.414451-08:00"
+    created: "2022-03-11T08:14:05.835958014-08:00"
     description: Azure disk Container Storage Interface (CSI) Storage Plugin
-    digest: 894d78bf1060ef20f53861ca0230a812449c2325463ac41b4d94689e1a10e996
+    digest: 7db62826b530290be444b3ce0010aa2cdbdb57a040c4bd537d5ae27edde72ed9
     name: azuredisk-csi-driver
     urls:
-    - https://raw.githubusercontent.com/kubernetes-sigs/azuredisk-csi-driver/main_v2/charts/latest-v2/azuredisk-csi-driver-v2.0.0-alpha.2.tgz
+    - https://raw.githubusercontent.com/abhisheksinghbaghel/azuredisk-csi-driver/master/charts/latest-v2/azuredisk-csi-driver-v2.0.0-alpha.2.tgz
```
Comment (edreed) [inline charts/index.yaml:11]: Use the following command to generate the index with the correct URL:

```sh
helm repo index ./charts --url  https://raw.githubusercontent.com/kubernetes-sigs/azuredisk-csi-driver/main_v2/charts
```

File: charts/latest-v2/azuredisk-csi-driver/values.yaml [hunk 2]
```diff
@@ -188,6 +189,7 @@ linux:
   tolerations:
     - operator: "Exists"
   hostNetwork: true # this setting could be disabled if perfProfile is `none`
+  getNodeInfoFromLabels: fakse # get node info from node labels instead of IMDS
```
Comment (edreed) [inline charts/latest-v2/azuredisk-csi-driver/values.yaml:192]: sp: `false`

File: charts/latest-v2/azuredisk-csi-driver/values.yaml [hunk 3]
```diff
@@ -221,6 +223,7 @@ windows:
   enabled: true
   dsName: csi-azuredisk-node-win # daemonset name
   kubelet: 'C:\var\lib\kubelet'
+  getNodeInfoFromLabels: fakse # get node info from node labels instead of IMDS
```
Comment (edreed) [inline charts/latest-v2/azuredisk-csi-driver/values.yaml:226]: sp: `false`

File: charts/latest-v2/azuredisk-csi-driver/templates/csi-azuredisk-node.yaml [hunk 4]
```diff
@@ -105,6 +105,8 @@ spec:
             - "--custom-user-agent={{ .Values.driver.customUserAgent }}"
             - "--user-agent-suffix={{ .Values.driver.userAgentSuffix }}"
             - "--allow-empty-cloud-config={{ .Values.node.allowEmptyCloudConfig }}"
+            - "--support-zone={{ .Values.node.supportZone }}"
+            - "--get-node-info-from-labels={{ .Values.node.getNodeInfoFromLabels }}"
```
Comment (edreed) [inline charts/latest-v2/azuredisk-csi-driver/templates/csi-azuredisk-node.yaml:109]: `.Values.linux.getNodeInfoFromLabels`

File: charts/index.yaml [hunk 5]
```diff
@@ -3,43 +3,70 @@ entries:
   azuredisk-csi-driver:
   - apiVersion: v1
     appVersion: latest-v2
-    created: "2022-02-28T11:03:23.414451-08:00"
+    created: "2022-03-14T12:48:19.443232748-07:00"
```
Comment (edreed) [inline charts/index.yaml:6]: Remember to rebuild the package and index when making changes to the chart.

File: docs/driver-parameters.md [hunk 6]
```diff
@@ -27,7 +27,7 @@ networkAccessPolicy | NetworkAccessPolicy property to prevent anybody from gener
 diskAccessID | ARM id of the DiskAccess resource for using private endpoints on disks | | No  | ``
 enableBursting | [enable on-demand bursting](https://docs.microsoft.com/en-us/azure/virtual-machines/disk-bursting) beyond the provisioned performance target of the disk. On-demand bursting only be applied to Premium disk, disk size > 512GB, Ultra & shared disk is not supported. Bursting is disabled by default. | `true`, `false` | No | `false`
 useragent | User agent used for [customer usage attribution](https://docs.microsoft.com/en-us/azure/marketplace/azure-partner-customer-usage-attribution)| | No  | Generated Useragent formatted `driverName/driverVersion compiler/version (OS-ARCH)`
-enableAsyncAttach | allow multiple disk attach operations (in batch) on one node in parallel, this could speed up disk attachment while may hit Azure API throttling when there are large number of volume attachments | `true`, `false` | No | `true`
+enableAsyncAttach | allow multiple disk attach operations (in batch) on one node in parallel, this could speed up disk attachment while may hit Azure API throttling when there are large number of volume attachments | `true`, `false` | No | `false`
```
Comment (nearora-msft) [inline docs/driver-parameters.md:30]: Should this param be added to the V2 section of the doc?

File: charts/latest/azuredisk-csi-driver/templates/rbac-csi-azuredisk-node.yaml [hunk 7]
```diff
@@ -8,6 +8,9 @@ rules:
   - apiGroups: [""]
     resources: ["secrets"]
     verbs: ["get", "list"]
+  - apiGroups: [""]
+    resources: ["nodes"]
+    verbs: ["get"]
```
Comment (edreed) [inline charts/latest/azuredisk-csi-driver/templates/rbac-csi-azuredisk-node.yaml:13]: We need to make these additions in the v2 chart.

File: pkg/azuredisk/nodeserver.go [hunk 8]
```diff
@@ -491,9 +519,21 @@ func (d *Driver) NodeExpandVolume(ctx context.Context, req *csi.NodeExpandVolume
 	if err != nil {
 		return nil, status.Errorf(codes.NotFound, "failed to determine device path for volumePath [%v]: %v", volumePath, err)
 	}
+	if !isBlock {
```
Comment (edreed) [inline pkg/azuredisk/nodeserver.go:522]: I don't see the changes below in `nodeserver_v2` or the `NodeProvisioner`.

File: pkg/provisioner/fake_nodeprovisioner.go [hunk 9]
```diff
@@ -104,3 +105,7 @@ func (fake *FakeNodeProvisioner) SetMountCheckError(target string, err error) {
 
 	fakeMounter.SetMountCheckError(target, err)
 }
+
+func (fake *FakeNodeProvisioner) SetMounter(mounter *mount.SafeFormatAndMount) {
```
Comment (edreed) [inline pkg/provisioner/fake_nodeprovisioner.go:109]: Where is this used?

Comment (mukhoakash) [reply pkg/provisioner/fake_nodeprovisioner.go:109]: `fake_azuredisk_v2.go` line: 165. Used to set the mounter through the fakeDriverV2.

File: pkg/azuredisk/nodeserver.go (hunk 8 above)
Comment (mukhoakash) [reply pkg/azuredisk/nodeserver.go:522]: Fixed

**Pull request merged**